library/pk.c

```diff
@@ -1569,6 +1569,8 @@ int mbedtls_pk_verify_restartable( mbedtls_pk_context *ctx,
     if( verify_ret == 0 )
     {
         mbedtls_platform_random_delay();
+        verify_ret = pk_info_verify_func( MBEDTLS_PK_CTX_INFO( ctx ),
```

Comment:
@AndrzejKurek , it is better to call twice the function from highest application level, in this case from ssl level. Is it possible? Why was the reason to call from this specific place ?

Reply:
I assumed that the fact that we are performing the verification twice is more of an implementation than the application choice, but you're right that current implementation will perform all verification twice (so, for example, the certificate will also be verified twice). 
I can move it to ssl_cli.c, lines ~3096, if that's the only thing we want: https://github.com/ARMmbed/mbedtls/blob/858e4325d2eb55274950e9335d0c08a1326069c2/library/ssl_cli.c#L3096

Reply:
there is also server side verify should be taken into account
https://github.com/ARMmbed/mbedtls/blob/858e4325d2eb55274950e9335d0c08a1326069c2/library/ssl_srv.c#L4645

Reply:
But generally, we want only from ssl level, these to places only should be protected. We also should verify that the operation itself was not skipped (but without adding code size  :) :) :) )

Reply:
Do you have any solution in mind? Isn't protecting the verification implementations enough (They'll return `PLATFORM_FAULT` if they find anything problematic)?